controller_manager/src/ros2_control_node.cpp

```diff
@@ -23,6 +24,40 @@
 
 using namespace std::chrono_literals;
 
+namespace
+{
+// Reference: https://man7.org/linux/man-pages/man2/sched_setparam.2.html
+// This value is used when configuring the main loop to use SCHED_FIFO scheduling
+// We use a midpoint RT priority to allow maximum flexibility to users
+int const kSchedPriority = 50;
+
+// Detect the presence of a RT kernel
+bool has_realtime_kernel()
+{
+  std::ifstream realtime_file("/sys/kernel/realtime", std::ios::in);
+  bool has_realtime = false;
+  if (realtime_file.is_open())
+  {
+    realtime_file >> has_realtime;
+  }
+  return has_realtime;
+}
+
+// Configure fifo sched for RT
+bool configure_sched_fifo(int priority)
+{
+  struct sched_param schedp;
+  memset(&schedp, 0, sizeof(schedp));
+  schedp.sched_priority = priority;
+  if (sched_setscheduler(0, SCHED_FIFO, &schedp))
```

Comment:
I really wanted to use SCHED_DEADLINE here but the problem is that the code to configure it does not compile without the headers from the RT kernel. For that reason I used FIFO which will compile on normal linux.

Reply:
Can we add this, maybe in another file or as the option somewhere users may use if they have RT kernel.

Can we use pre-processor directive to detect that and set different scheduler if possible?

Reply:
1. Yes, I'll move these into a separate file with a header so others can easily use them in their code.
2. You wouldn't want to do that because then binaries built in the buildfarm wouldn't have this feature.